internal/controller/progressive/progressive.go

```diff
@@ -482,16 +479,13 @@ func checkForUpgradeReplacement(
 
 	if differentFromExistingUpgrading {
 
-		// prepare existing upgrading child for Recycle
-		err = controller.ProcessUpgradingChildPreRecycle(ctx, rolloutObject, existingUpgradingChildDef, c)
-		if err != nil {
-			return false, false, err
-		}
-
 		if differentFromPromoted {
 
 			// recycle the old one
 			reason := common.LabelValueProgressiveReplaced
+			if childStatus.AssessmentResult == apiv1.AssessmentResultFailure {
+				reason = common.LabelValueProgressiveReplacedFailed
+			}
```

Comment:
I wanted to distinguish the reason if there was a Failure that was assessed. In this case, we can optimize by not attempting to drain with original spec, since we know it will not work.